doc/Language/rb-nutshell.pod

```diff
@@ -0,0 +1,944 @@
+    our $foo = 7   # Perl 6, package scoped
+    has $!foo = 7  # Perl 6, instance scoped (attribute)
+
+
+=head2 Operators
+
+See S03-operators for full details on all operators.
+
+Unchanged:
+
+=item C<,> List Separator
+=item C<+> Numeric Addition
+=item C<-> Numeric Subtraction
+=item C<*> Numeric Multiplication
+=item C</> Numeric Division
+=item C<%> Numeric Modulus
+=item C<**> Numeric Exponentiation
+=item C<! && || ^> Booleans, high-precedence
+=item C<not and or xor> Booleans, low-precedence
+=item C«== != < > <= >=»   Numeric comparisons
+
+=head3 C<++> Numeric Increment
+=head3 C<--> Numeric Decrement
+
+You may use C<$x++> instead of C<x += 1> as a shortcut for incrementing a
+variable. This can be used as a pre-increment C<++$x> (increment, return new
+value) or post-increment C<$x++> (increment, return old value).
+
+You may use C<$x--> instead of C<x -= 1> as a shortcut for decrementing a
+variable. This can be used as a pre-decrement C<--$x> (decrement, return new
+value) or post-decrement C<$x--> (decrement, return old value).
+
+=head3 C<eq ne lt gt le ge> String comparisons
+
+String comparisions are separated from numeric comparisons to make explicit
+what sort of comparison you are doing. Using C<==> tries to convert the values
+to numbers, and C<eq> tries to convert the values to strings.
+
+=head3 C«<=> leg» Three-way comparisons
+
+In Ruby, the C«<=>» operator returns -1, 0, or 1.
+In Perl 6, they return C<Order::Less>, C<Order::Same>, or C<Order::More>.
+
+C«<=>» forces numeric context for the comparison.
+
+C«leg» ("Less, Equal, or Greater?") forces string context for the comparison.
+
+C«cmp» does either C«<=>» or C<leg>, depending on the existing type of its
+arguments.
+
+=head3 C<~~> Smart-match operator
+
+This is a very common matching operator. Here are some examples:
+
+    say "match!" if $foo ~~ /bar/;      # Regex match
+    say "match!" if $foo ~~ "bar";      # String match
+    say "match!" if $foo ~~ :(Int, Str) # Signature match (destructure)
+
+See L<S03/Smart matching|https://design.perl6.org/S03.html#Smart_matching>
+
+=head3 C<& | ^> Numeric Bitwise ops
+=head3 C<& | ^> Boolean ops
+
+In Perl 6, these single-character ops have been removed, and replaced by
+two-character ops which coerce their arguments to the needed context.
+
+    # Infix ops (two arguments; one on each side of the op)
+    +&  +|  +^  And Or Xor: Numeric
+    ~&  ~|  ~^  And Or Xor: String
+    ?&  ?|  ?^  And Or Xor: Boolean
+
+    # Prefix ops (one argument, after the op)
+    +^  Not: Numeric
+    ~^  Not: String
+    ?^  Not: Boolean (same as the ! op)
+
+=head3 C<&.> Conditional chaining operator
+
+Ruby uses the C<&.> operator to chain methods without raising an error if one
+invocation returns nil. In Perl 6 use C<.?> for that.
+
+=head3 C«<< >>» Numeric shift left|right ops
+
+Replaced by C«+<» and C«+>» .
+
+    puts 42 << 3  # Ruby
+    say  42 +< 3; # Perl 6
```

Comment:
Add a note about how some Ruby classes override the << operator to mean roughly "push". This is called the "shovel operator" in Ruby